test/integration/targets/docker-registry/tasks/main.yml

```diff
@@ -0,0 +1,139 @@
+
+- block:
+  - name: Start test registry
+    docker_container:
+      name: "{{ registry_name }}"
+      image: registry:2.6.1
+      ports: 5000
+    register: registry_container
+
+  - name: Get registry URL
+    set_fact:
+      registry_address: "localhost:{{ registry_container.container.NetworkSettings.Ports['5000/tcp'].0.HostPort }}"
+
+  - name: Start nginx frontend for registry
+    docker_volume:
+      name: "{{ nginx_name }}"
+      state: present
+
+  - name: Create container for nginx frontend for registry
+    docker_container:
+      state: stopped
+      name: "{{ nginx_name }}"
+      image: nginx:alpine
+      ports: 5000
+      links:
+        - "{{ registry_name }}:real-registry"
+      volumes:
+        - "{{ nginx_name }}:/etc/nginx/"
+    register: nginx_container
+
+  - name: Copy static files into volume
+    command: docker cp {{ role_path }}/files/{{ item }} {{ nginx_name }}:/etc/nginx/{{ item }}
+    loop:
+      - "nginx.conf"
+      - "nginx.htpasswd"
+
+  - name: Create private key for frontend certificate
+    openssl_privatekey:
+      path: "{{ output_dir }}/cert.key"
+      type: ECC
+      curve: secp256r1
+  - name: Create CSR for frontend certificate
+    openssl_csr:
+      path: "{{ output_dir }}/cert.csr"
+      privatekey_path: "{{ output_dir }}/cert.key"
+      subject_alt_name:
+        - "DNS:test-registry.ansible.com"
+  - name: Create frontend certificate
+    openssl_certificate:
+      path: "{{ output_dir }}/cert.pem"
+      csr_path: "{{ output_dir }}/cert.csr"
+      privatekey_path: "{{ output_dir }}/cert.key"
+      provider: selfsigned
+
+  - name: Copy dynamic files into volume
+    command: docker cp {{ output_dir }}/{{ item }} {{ nginx_name }}:/etc/nginx/{{ item }}
+    loop:
+      - "cert.pem"
+      - "cert.key"
+
+  - name: Start nginx frontend for registry
+    docker_container:
+      name: "{{ nginx_name }}"
+      state: started
+      restart: yes  # in case it was already running
+    register: nginx_container
+
+  - debug: var=nginx_container.container.NetworkSettings
+
+  - name: Wait for registry frontend
+    uri:
+      url: "https://{{ nginx_container.container.NetworkSettings.IPAddress }}:5000/v2/"
+      url_username: testuser
+      url_password: hunter2
+      validate_certs: no
+    register: result
+    until: result is success
+    retries: 5
+    delay: 1
+
+  - name: Get registry URL
+    set_fact:
+      registry_frontend_address: "localhost:{{ nginx_container.container.NetworkSettings.Ports['5000/tcp'].0.HostPort }}"
+
+  - debug: msg="Registry available under {{ registry_address }}, NGINX frontend available under {{ registry_frontend_address }}"
+
+  - include_tasks: run-test.yml
+    with_fileglob:
+    - "tests/*.yml"
+
+  always:
+  - name: "Make sure all images are removed"
+    docker_image:
+      name: "{{ item }}"
+      state: absent
+    with_items: "{{ inames }}"
+  - name: "Get registry logs"
```

Comment:
Note that due to the default ansible log level used in shippable runs, both docker and nginx logs are printed twice.

Reply:
That's true for shippable, but not for local runs. (If #24215 would be implemented it would be easier to work around this problem :) )

Reply:
`no_log` could be added to the command tasks.

Reply:
Thanks, added that.